src/graphql/graphiql.js

```diff
@@ -0,0 +1,130 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Code related to GraphiQL
+
+import { parse } from 'graphql'
+import { createGraphQLUrls } from '@/utils/graphql'
+import store from '@/store'
+
+// TODO: https://github.com/apollographql/GraphiQL-Subscriptions-Fetcher/issues/16
+//       the functions hasSubscriptionOperation and graphQLFetcher are both from
+//       the graphiql-subscriptions-fetcher. Unfortunately that project is archived
+//       on GitHub, and is using the old API for subscription-transport-ws, which
+//       is a dependency of Cylc UI. As we cannot use an older version, instead we
+//       have the two functions here, patched as per issue to work with newer API.
+
+/**
+ * Tell whether it is a query or subscription.
+ *
+ * @private
+ * @param {{
+ *   query: string
+ * }}graphQlParams
+ * @returns {boolean} true if the params contain a subscription, false otherwise
+ */
+const hasSubscriptionOperation = function (graphQlParams) {
+  const queryDoc = parse(graphQlParams.query)
+  for (let _i = 0, _a = queryDoc.definitions; _i < _a.length; _i++) {
+    const definition = _a[_i]
+    if (definition.kind === 'OperationDefinition') {
+      const operation = definition.operation
+      if (operation === 'subscription') {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+/**
+ * The GraphQL fetcher function.
+ *
+ * @param {Object|null} subscriptionsClient
+ * @param {function} fallbackFetcher
+ * @returns {function(...[*]=)}
+ */
+const graphQLFetcher = function (subscriptionsClient, fallbackFetcher) {
+  let activeSubscription = null
+  return function (graphQLParams) {
+    if (subscriptionsClient && activeSubscription !== null) {
+      subscriptionsClient.unsubscribe(activeSubscription)
+    }
+    if (subscriptionsClient && hasSubscriptionOperation(graphQLParams)) {
+      return {
+        subscribe: function (observer) {
+          observer.next('Your subscription data will appear here after server publication!')
+          const subscription = subscriptionsClient.request({
+            query: graphQLParams.query,
+            variables: graphQLParams.variables
+          }, function (error, result) {
+            if (error) {
+              observer.error(error)
+            } else {
+              observer.next(result)
+            }
+          })
+          activeSubscription = subscription.subscribe((result, err) => {
```

Comment:
This is what changed. `.subscribe` here is **not** from GraphQL. This is the method of a JS subscription, that will create an Observer. Later we are destroying it, as leaving Observers hanging around leads to memory leaks.